Framework/Geometry/src/Objects/ShapeFactory.cpp

```diff
@@ -942,6 +926,36 @@ ShapeFactory::parseHexahedron(Poco::XML::Element *pElem,
   hex.rbb = parsePosition(pElem_rbb);
   hex.rbt = parsePosition(pElem_rbt);
 
+  const bool isValid = pElem_lfb && pElem_lft && pElem_lbb && pElem_lbt &&
+                       pElem_rfb && pElem_rft && pElem_rbb && pElem_rbt;
+
+  const std::string ERROR_MSG =
```

Comment:
Could we be even more verbose about this. One thing I dislike about the current implementations we have in Mantid is that they often don't provide enough information to track down problems. We could for example determine which pointer is null (not provided)